ntoskrnl/config/cminit.c

```diff
@@ -545,14 +545,14 @@ CmpOpenHiveFiles(IN PCUNICODE_STRING BaseName,
 
     /* Now create the file */
     Status = ZwCreateFile(Log,
-                          DesiredAccess,
+                          DesiredAccess | SYNCHRONIZE,
```

Comment:
Why is this needed? (same question for the IoFlags too)

Reply:
Without this flag, it have stack corrupt. It happen if inside RegistryHive->FileWrite function ZwWriteFile return STATUS_PENDING. Or in RegistryHive->FileFlush function ZwFlushBuffersFile return STATUS_PENDING.

Reply:
Does that claim still stand @mrmks04 with todays master head? Again see the already merged https://github.com/reactos/reactos/pull/3679

Reply:
With latest code from master problem gone.